plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java

```diff
@@ -258,4 +272,38 @@ public void validate()
                 !countDistinctPushdownEnabled || aggregationPushdownEnabled,
                 "Invalid configuration: pinot.aggregation-pushdown.enabled must be enabled if pinot.count-distinct-pushdown.enabled");
     }
+
+    @AssertTrue(message = "Controller URL should use HTTPS scheme when TLS is enabled")
```

Comment:
Why not just decide whether to use TLS or not based on the url scheme. It seems better than having two flags that need to agree.